tests/test_base.py

```diff
@@ -4,3 +4,26 @@
 def test_import():
     import qutip_cupy
     assert qutip_cupy.__version__
+
+def test_conversion_cycle():
+    from qutip_cupy import CuPyDense
+    from qutip_cupy import data
+
+    old_dense = data.Dense([[0,1,2,6,7,8]])
+
+    tr1 = data.to[CuPyDense, data.Dense](old_dense)
+
+    assert (old_dense.to_array() == data.to[data.Dense, CuPyDense](tr1).to_array()).all()
```

Comment:
The normal form of `data.to` is `data.to(array, type)` as a function - it's easier to read, and the function will infer what it needs.

Reply:
Nice, it looks a lot better this way.